geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -192,40 +194,35 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
     // index api defines start row inclusive, end row exclusive
     // both these conventions match the conventions for hbase scan objects
     val ranges = byteRanges.map {
-      case BoundedByteRange(start, end) => new Scan(start, end)
-      case SingleRowByteRange(row)      => new Scan(row, ByteArrays.rowFollowingRow(row)).setSmall(true)
+      case BoundedByteRange(start, stop) => new RowRange(start, true, stop, false)
+      case SingleRowByteRange(row)       => new RowRange(row, true, ByteArrays.rowFollowingRow(row), false)
     }
+    val small = byteRanges.headOption.exists(_.isInstanceOf[SingleRowByteRange])
```

Comment:
Ah, does this cover the case where we are doing a point lookup in, say, an FID index?

Reply:
right